### DISCUSSION TITLE
DataGridComboBox - Binding to an Object

### DISCUSSION THREAD

**Discussion by amrit7** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3550] [Q&A] [answered]
Hi,

I'm using a DataGrid and trying to bind to an Object - it is not working - any ideas?
I can bind to a string List correctly.

I've created a very small project to demonstrate what I'm trying - with a few different Combo Boxes - some working with string List Binding, some not with Object Binding:
https://github.com/amrit7/WPFDataGridComboBox

I would like a ComboBox to bind to an Object so that it can Display a Name and also hold an ID. 
Then when I change the Name on the ComboBox, I want the ID to be able to update a table correctly.

Any ideas?

**Comment by MichelMichels** [comment, marked as answer]
@amrit7 This likely has nothing to do with MDIX, but rather the lack of implementing `INotifyPropertyChanged` for the list property bound to the ComboBox ItemSource. The other lists work because you initialize them immediately where as the list where it doesn't work, gets filled in the constructor.
